pkg/resource/replication_group/custom_update_api.go

```diff
@@ -70,6 +73,22 @@ func (rm *resourceManager) CustomModifyReplicationGroup(
 			requeue.DefaultRequeueAfterDuration)
 	}
 
+	// Handle the asynchronous rollback case for while Scaling down.
+	// This means that we have already attempted to apply the CacheNodeType once and
+	// were not successful hence we will set a terminal condition.
+	if !cacheNodeTypeRequiresUpdate(desired) && delta.DifferentAt("Spec.CacheNodeType") {
+		return nil, awserr.New("InvalidParameterCombination", "Cannot update CacheNodeType, "+
```

Comment:
Is there a reason why you are using `awserr.New` and not `errors.New` ?

Reply:
This handles setting terminal codes correctly. Terminal codes are set using `awsErr.Code()`